lib/bolt/transport/ssh/connection.rb

```diff
@@ -275,6 +312,32 @@ def validate_ssh_version
             raise "Powershell over SSH requires OpenSSH server >= 7.9, target is running #{match[1]}"
           end
         end
+
+        def handle_net_ssh_error(err)
+          match = /could not settle on (?<algo>encryption|hmac|host_key|kex)/.match(err.message)
+
+          if match
+            enabled = Net::SSH::Transport::Algorithms::DEFAULT_ALGORITHMS[match[:algo].to_sym]
```

Comment:
I think 'available' and 'deprecated' would be better words for these. They aren't what's configured by the user, and I think 'enabled' indicates something on the user-side.